src/shared/components/setup-locker/setup-locker-steps/set-timeout/SetTimeout.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import { IdlePicker, Button } from '@nomios/web-uikit';
+import PropTypes from 'prop-types';
+
+import styles from './SetTimeout.css';
+
+class SetTimeout extends Component {
+    state = {
+        timeoutValue: 3,
+    };
+
+    render() {
+        const { timeoutValue, feedback } = this.state;
+
+        return (
+            <div className={ styles.contentWrapper }>
+                <h2 className={ styles.title }>Setup expiration time.</h2>
+                <p>You may tweak the amount of time you are ablo to use
+                 the app before it locks again. You can go have a coffee without any worries.</p>
+                <div className={ styles.idlePicker }>
+                    <IdlePicker defaultValue={ timeoutValue } handlePickerChange={ this.onPickerChange } />
+                </div>
+                <div className={ styles.continueButton }>
+                    <Button onClick={ this.handleContinue } onAnimationEnd={ this.handleButtonAnimationEnd }
+                        feedback={ feedback }>Finish</Button>
+                </div>
+            </div>
+        );
+    }
+
+    onPickerChange = (value) => {
+        this.setState({ timeoutValue: value });
+    };
+
+    handleContinue = async () => {
+        const { timeoutValue } = this.state;
+        const { setMaxTime } = this.props;
+
+        try {
+            await setMaxTime(timeoutValue * 60 * 1000);
```

Comment:
We should be careful when doing things like this: what happens if the component is unmounted while the promise is still resolving? If that happens, the `setState` will be called on a unmounted component.

Reply:
What would be the best approach to prevent that scenario?